### DISCUSSION TITLE
is data property removed from the supabase.auth.signUp

### DISCUSSION THREAD

**Discussion by HassanIyan** [supabase/supabase#3782] [Questions] [answered]
let { error } = await supabase.auth.signUp(
                {
                    email: email,
                    password: password,
                    data: {
                        fName: fName,
                        lName: lName
                    }
                }
            )

i use the above code to add user meta data. a little while back with this method i can add user meta data. and can access it with supabase.auth.user().user_metadata

unfortunately, now with this code it is not adding the data to database. and user_meta data returns a empty array.

also while going through this i found out that is vscode email, password, provider, phone and every thing are highlighted as a property of supabase.auth.signup, but data is not. so did they remove the data property??

**Comment by dshukertjr** [comment, marked as answer]
Hi @HassanIyan!

It seems like `data` is still alive and well [here](https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L115). Not sure why it stopped working for you. That being said, it depends on the specific use case, but it is generally better to store user information such as first names and last names on a `public.users` table as suggested [here](https://supabase.io/docs/guides/auth/managing-user-data), so maybe you could consider this option!
